### ISSUE TITLE
Removal of elements from DOM after <dom-repeat> filter is not chunked

### ISSUE THREAD

**Issue by thomastodon** [Polymer/polymer#4729]
### Description
On initial render, `<dom-repeat>` renders are chunked via element attributes: `target-framerate` and `initial-count`. When I attempt to filter the elements via `<dom-repeat>`'s `filter` attribute, it does not remove the _filtered-out_ elements from the DOM in a _chunked_ manner. It blocks the main thread until everything is removed.

#### Demo
The below demo observes on the `dom-change` event to demonstrate the issue:
http://jsbin.com/cugawafusu/edit?html,console,output

#### Steps to Reproduce
1. Load the page in the demo above.
2. Observe _"Mutating the DOM!"_ logged in the console **ten** times, once for each of the `items` in the `<dom-repeat>`. 
3. Press the button marked **Click to Filter**

#### Expected Results
_"Mutating the DOM!"_ is logged in the console **five** additional times. Once for each element removed from the DOM.

#### Actual Results
_"Mutating the DOM!"_ is only logged in the console **one** additional time.

### Browsers Affected
- [x] Chrome
- [ ] Firefox
- [ ] Edge
- [ ] Safari 9
- [ ] Safari 8
- [ ] IE 11

_I don't believe this is browser specific_

### Versions
- Polymer: v2.0.1
- webcomponents: v1.0.0

**Comment by thomastodon** [comment]
I made a fix on **2.0.1**, which appears to already have been committed on **master**. For reference:
```
--- a/lib/elements/dom-repeat.html
+++ b/lib/elements/dom-repeat.html
@@ -435,6 +435,7 @@ subject to an additional IP rights grant found at http://polymer.github.io/PATEN
       if (!this.__handleItemPath(change.path, change.value)) {
         // Otherwise, the array was reset ('items') or spliced ('items.splices'),
         // so queue a full refresh
-        this.__needFullRefresh = true;
         this.__initializeChunking();
         this.__debounceRender(this.__render);
       }
```

**Comment by sorvell** [comment]
Chunking is currently only implemented when adding items and not removing them. It would be good to add removal to chunking. @kevinpschaaf wdyt?
